runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/request/HttpRequestBuilder.kt

```diff
@@ -36,6 +36,35 @@ public class HttpRequestBuilder private constructor(
     }
 }
 
+internal data class ImmutableHttpRequestBuilder(
+    internal val builder: HttpRequestBuilder,
+    internal val allowToBuilder: Boolean,
+) : HttpRequest {
```

Comment:
**Nit**: This class doesn't feel like a "builder". It's not externally mutable and may be turned into a builder by the `toBuilder()` extension method. Maybe it's an `ImmutableHttpRequest` or an `HttpRequestBuilderView`?